File: Pascal/chiper.pas
program caesar_chiper;

uses crt;

var 
    text : string ;
    key  : integer;


    function encrypt(text:string; key:integer) : string;

    var
        i, k : integer;
        res : string;
    begin
        res := '';
        for i := 1 to length(text) do
            begin
            k := key mod 25;
            if ((ord(text[i]) >= 97) and (ord(text[i]) <= 122)) then 
                begin
                    if (ord(text[i]) + k > 122) then
                        begin
                        k := ord(text[i]) + k - 122;
                        res := res + chr(97 + k)
                        end
                    else
                        res :=  res + chr(ord(text[i]) + k)
                end
            else if ((ord(text[i]) >= 65) and (ord(text[i]) <= 90)) then 
                begin
                    if (ord(text[i]) + k > 90) then
                        begin
                        k := ord(text[i]) + k - 90;
                        res := res + chr(65 + k)
                        end
                    else
                        res :=  res + chr(ord(text[i]) + k)
                end
            else
                res :=  res + text[i];
            end;
    
        encrypt := res

    end;

begin
    clrscr();
    readln(text);
    readln(key);
    writeln(encrypt(text, key));
end.

File: Go/chiper.go
package main

import "fmt"

func encrypt(text string, key int) string {
	res := ""

	for _, letter := range text {
		var k = key % 25
		if (letter >= 97 && letter <= 122) {
			if (int(letter) + k > 122) {
				k = int(letter) + k - 122
				res += string(97 + k)
			} else {
				res +=  string(int(letter) + k) 
			}
		} else if (letter >= 65 && letter <= 90) {
			if (int(letter) + k > 90) {
				k = int(letter) + k - 90
				res += string(65 + k)
			} else {
				res +=  string(int(letter) + k) 
			}
		} else {
			res += string(letter)
		}
	}

	return res
}

func main() {
	var text string = ""
	fmt.Scanln(&text)
	var key int = 0
	fmt.Scanln(&key)
	fmt.Println(encrypt(text, key))
}
